client/src/main/java/org/eclipse/hono/client/ProtocolAdapterCommandConsumerFactory.java

```diff
@@ -83,6 +85,7 @@ void initialize(CommandTargetMapper commandTargetMapper,
             String tenantId,
             String deviceId,
             Handler<CommandContext> commandHandler,
+            int lifespanSeconds,
```

Comment:
I would prefer the use of `Duration` rather then `int`, which prevents any confusion around the unit.

Reply:
I've amended the commit, changing the type to `Duration` (also in the `DeviceConnectionInfo`, `DeviceConnectionClient` and `DeviceConnectionService` interfaces committed earlier).
I've opted against the usage of `Optional` (it produces quite some clutter here, IMHO). The way to indicate an infinite lifespan is now using either `null` or a negative `Duration`.